### PR TITLE
test(test-runner): add test for hiding conditional expect matchers

### PR DESCRIPTION
**Pull request by mxschmitt**
Added a test for #10016 and #10670

### CODE REVIEW

File: tests/playwright-test/expect.spec.ts [hunk 1]
```diff
@@ -50,6 +50,23 @@ test('should be able to call expect.extend in config', async ({ runInlineTest })
   expect(result.passed).toBe(1);
 });
 
+test('should not expand huge arrays', async ({ runInlineTest }) => {
```
Comment (mxschmitt) [inline tests/playwright-test/expect.spec.ts:53]: drive-by: moved above the `runTSC` tests so its grouped.

File: tests/playwright-test/expect.spec.ts [hunk 2]
```diff
@@ -126,19 +143,29 @@ test('should work with custom PlaywrightTest namespace', async ({ runTSC }) => {
...
+    'a.spec.ts': `
+    const { test } = pwt;
+    test('custom matchers', async ({ page }) => {
+      await test.expect(page).toHaveURL('https://example.com');
+      await test.expect(page).toBe(true);
+      // @ts-expect-error
+      await test.expect(page).toBeEnabled();
+
+      await test.expect(page.locator('foo')).toBeEnabled();
+      await test.expect(page.locator('foo')).toBe(true);
```
Comment (dgozman) [inline tests/playwright-test/expect.spec.ts:158]: I'd think that this one is also an error.

**Review by dgozman — APPROVED** [review]

File: tests/playwright-test/expect.spec.ts (hunk 2 above)
Comment (mxschmitt) [reply tests/playwright-test/expect.spec.ts:160]: discussed offline: lets hide the default expect matchers for now. -> applied suggestion.
